org.bitcoinknots.bitcoin-qt.json

```diff
@@ -0,0 +1,91 @@
+        "--socket=pulseaudio",
+        "--device=dri",
+        "--share=ipc",
+        "--share=network",
+        "--filesystem=/run/tor/control.authcookie:ro"
+    ],
+    "modules": [
+        {
+            "name": "bitcoinknots",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -m 0744 -D -t /app/bin/internal bin/bitcoind",
+                "install -m 0744 -D -t /app/bin/internal bin/bitcoin-qt",
+                "install -m 0744 -D -t /app/bin/internal bin/bitcoin-cli",
+                "install -m 0744 -D -t /app/bin/internal bin/test_bitcoin"
+            ],
+            "post-install": [
+                "/app/bin/internal/test_bitcoin"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://bitcoinknots.org/files/26.x/26.1.knots20240325/bitcoin-26.1.knots20240325-x86_64-linux-gnu.tar.gz",
+                    "sha256": "ae13675c1b66ce7691f367ffa412f6146af0d951e1ff95a3952e31f34bb25b94",
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://bitcoinknots.org/",
+                        "version-pattern": "Latest version: ([\\d\\.-]+\\.knots\\d+)",
+                        "url-template": "https://bitcoinknots.org/files/$major.x/$version/bitcoin-$version-x86_64-linux-gnu.tar.gz"
+                    }
+                },
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "aarch64"
+                    ],
+                    "url": "https://bitcoinknots.org/files/26.x/26.1.knots20240325/bitcoin-26.1.knots20240325-aarch64-linux-gnu.tar.gz",
+                    "sha256": "39cca43d82fc9482e2880f75093fb9e8512c1c62383954e33a7b7a7f8ea4ae9a",
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://bitcoinknots.org/",
+                        "version-pattern": "Latest version: ([\\d\\.-]+\\.knots\\d+)",
+                        "url-template": "https://bitcoinknots.org/files/$major.x/$version/bitcoin-$version-aarch64-linux-gnu.tar.gz"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "misc",
+            "buildsystem": "simple",
+            "build-commands": [
+                "desktop-file-edit --set-key=\"Icon\" --set-value=${FLATPAK_ID} bitcoin-qt.desktop",
+                "install -Dp -m 644 bitcoin-qt.desktop /app/share/applications/${FLATPAK_ID}.desktop",
+                "install -Dp -m 644 bitcoin.svg /app/share/icons/hicolor/scalable/apps/org.bitcoinknots.bitcoin-qt.svg",
+                "install -Dp -m 644 org.bitcoinknots.bitcoin-qt.metainfo.xml /app/share/metainfo/org.bitcoinknots.bitcoin-qt.metainfo.xml",
+                "bash install_wrappers.sh"
```

Comment:
this does not need to be a separate shell script.

Reply:
Yet it is currently. I'm not trying to rewrite this, just port it over for now.